archive/import.go

```diff
@@ -287,16 +267,22 @@ func newChunkWriter(ctx context.Context, ark *Archive, tsDir tsDir, chunk Chunk)
 	}))
 	return &chunkWriter{
 		ark:             ark,
-		chunk:           chunk,
 		dataFileWriter:  dataFileWriter,
+		id:              id,
 		indexBuilder:    indexBuilder,
 		indexTempPath:   indexTempPath,
 		indexTempWriter: indexTempWriter,
+		kind:            kind,
+		masks:           masks,
 		needIndexWrite:  true,
-		tsDir:           tsDir,
+		tsd:             tsd,
 	}, nil
 }
 
+func (cw *chunkWriter) position() (int64, nano.Ts, nano.Ts) {
+	return cw.dataFileWriter.Position(), cw.firstTs, cw.lastTs
+}
```

Comment:
What are the constraints on this triple? Will the .Position() be between cw.firstTs and cw.lastTs?

Reply:
The Position() return parameter is the current offset in the writer, and is used to cap the length of the chunk that's being written by the chunkWriter. The firstTs value is the timestamp of the first record written to the chunk, and the lastTs value is the timetamp of the most recent record. So over the course of calling the `Write(*zng.Record)` method on chunkWriter, firstTs will stay the same, but Position() and lastTs will reflect the most recent Write call.